packages/engine/src/world/SketchManager.ts

```diff
@@ -12,9 +13,25 @@ type SketchInstance = {
   getPasses?: (engineScene: EngineScene) => Pass[]
 }
 
+let instance: SketchManager
```

Comment:
I really dislike all this (the class needing to be passed the engine, making it a singleton to access it in a way that works with the exported function), though it's better than needing an engine ref elsewhere. 

I tried going down the route of a create* function, but I hit a wall with needing to both call a function on the sketch instance, and make a change to the state

Reply:
I think we can solve this with React context. We could have an engine provider and then you can get the engine by doing something like `const engine = useHedronEngine`